CMSIS/RTOS2/FreeRTOS/Source/cmsis_os2.c

```diff
@@ -576,6 +575,7 @@ osThreadId_t osThreadNew (osThreadFunc_t func, void *argument, const osThreadAtt
     }
     else {
       mem = 0;
+      name = NULL;
```

Comment:
Please move this back to its original location (line 537).
I know `name` is assigned two times when attributes are provided but the whole module tends to initialize variables somewhere at the top and modifies them later after some decision making. It just makes things simple and consistent - and we can survive extra CPU instructions when building without optimization.